### PR TITLE
Bug 1232589 - wrap Cellular And Data panel into AMD module, r=yzen

### PR DESCRIPTION
**Pull request by gasolin**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0b37dbb590ec275e44923ffdcdce5d63f6e9d64d

File: apps/settings/elements/carrier.html [hunk 1]
```diff
@@ -70,9 +70,7 @@ <h2 data-l10n-id="simSettings"></h2>
       </div>
     </div>
 
-    <script src="shared/js/apn_helper.js"></script>
-    <script src="shared/js/settings_helper.js"></script>
```
Comment (gasolin) [inline apps/settings/elements/carrier.html]: `apn_helper` and `settings_helper` are not used anymore

File: apps/settings/js/panels/carrier/panel.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ */
+define(function(require) {
+  'use strict';
+
+  var SettingsPanel = require('modules/settings_panel');
+  var Carrier = require('panels/carrier/carrier');
+
+  return function ctor_media_storage_panel() {
+    return SettingsPanel({
+      onInit: function(panel) {
```
Comment (yzen) [inline apps/settings/js/panels/carrier/panel.js]: Nit: lets drop an unused panel argument.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d3ec179f4731b78b6abd3696252b0f3b827e49b8

**Pull request merged**